proxy/src/transport/tls/config.rs

```diff
@@ -249,15 +249,27 @@ impl CommonConfig {
 
 }
 
+// A Future that, when polled, checks for config updates and publishes them.
+pub type PublishConfigs = Box<Future<Item = (), Error = ()> + Send>;
+
+/// Returns Client and Server config watches, and a task to drive updates.
+///
+/// The returned task Future is expected to never complete.
+///
+/// If there are no TLS settings, then empty watches are returned. In this case, the
+/// Future that is never notified.
+///
+/// If all references are dropped to _either_ the client or server config watches, all
+/// updates will cease for both config watches.
 pub fn watch_for_config_changes(settings: Conditional<&CommonSettings, ReasonForNoTls>)
-    -> (ClientConfigWatch, ServerConfigWatch, Box<Future<Item = (), Error = ()> + Send>)
+    -> (ClientConfigWatch, ServerConfigWatch, PublishConfigs)
 {
     let settings = if let Conditional::Some(settings) = settings {
         settings.clone()
     } else {
         let (client_watch, _) = Watch::new(None);
         let (server_watch, _) = Watch::new(None);
-        let no_future = future::ok(());
+        let no_future = future::empty();
```

Comment:
I initially thought it was better for the future to finish immediately when there's no work for it to do, so the executor doesn't have to waste time polling a future that we don't care about, but perhaps this doesn't matter?

Reply:
It doesn't spend any time polling it if it's not notified ;)